api/controllers/Irc_EntryController.js

```diff
@@ -63,22 +63,38 @@ module.exports = {
   },
 
   process_event: function (request_body) {
-    if (request_body.operation === 'check in' || request_body.operation === 'update individual') {
-      return this.processEventDetaineeCreateOrUpdate(request_body);
+    if (request_body.operation === 'check in') {
+      return this.saveEvent(request_body).then(this.saveDetainee);
     }
     throw new ValidationError('Unknown');
   },
 
+  getPID: function (entity) {
+    return `${entity.centre}_${entity.person_id}`;
+  },
 
-  processEventDetaineeCreateOrUpdate: (request_body) => {
-    return Subjects.findOrCreate({
-      person_id: request_body.person_id
+  saveEvent: function (request_body) {
+    var person_id = this.getPID(request_body);
+    return Events.findOrCreate({
+      person_id: person_id
```

Comment:
I think you just want to create an event everytime rather than potentially find one first?
if someone checked in more than once you'd only ever have the first one?